### PR TITLE
Afd 106 book details logic

### PR DESCRIPTION
**Pull request by TheGuardingDark**
Clicking on a book in the bookshelf will display a details page with the books stored info displayed. A delete button allows you to delete the book and an edit button will take you to an edit page that has not been set-up yet.

Changed way dates are displayed, added delete button and functionality, added to the stars logic to be disabled when on the details page. No way to add book jacket images at this time

known issues: bottom margin at 769px

### CODE REVIEW

**Comment by katieburrows** [comment]
When I added the book it didn't seem to go to the bookshelf, but I don't believe that functionality is within the scope of this ticket.  One idea for refactoring is that when a user is on the Edit Page and tabbing through the form, the focus is out of order.  So instead of title -> author -> synopsis it goes title -> author -> add image button.

**Review by katieburrows — APPROVED** [review]
Please see comment re:  refactoring, otherwise, looking good!

File: src/components/Stars.jsx [hunk 1]
```diff
@@ -11,26 +23,37 @@ const Stars = ({ handleRatingChange, reset, setReset }) => {
...
-                givenRating <= currentRating || givenRating === currentRating
-                  ? 'form__rating--checked'
-                  : 'form__rating--unchecked'
-              }
-              onClick={() => {
-                handleRatingChange(givenRating);
-                setCurrentRating(givenRating);
-              }}
-            />
+            {path === detailsPath ? (
```
Comment (scullenDev) [inline src/components/Stars.jsx:31]: Clever handling of read-only vs settable star rating!

File: src/components/Stars.jsx [hunk 2]
```diff
@@ -11,26 +23,37 @@ const Stars = ({ handleRatingChange, reset, setReset }) => {
...
+              <FaStar
+                style={{ cursor: 'default' }}
+                className={
+                  givenRating <= bookRating || givenRating === bookRating
```
Comment (scullenDev) [inline src/components/Stars.jsx:35]: So, for a second I was like, "Whoaaaaa, is that a reverse arrow function?! What sorcery is this???!!!"

Comment (scullenDev) [reply src/components/Stars.jsx:35]: But then I remembered less-than-or-equal to is a thing and I laughed out loud at myself.

File: src/pages/Details.jsx [hunk 3]
```diff
@@ -1,39 +1,59 @@
...
+        Published: {published ? `${month}/${day}/${year}` : ''}
+      </h3>
+      <h3 className="details__text details__text--pages">
+        {pages || ''} Pages
       </h3>
-      <h3 className="details__text details__text--pages">{pages} Pages</h3>
       <p className="details__text details__text--synopsis">
         <q>{synopsis}</q>
       </p>
+      <Button
```
Comment (scullenDev) [inline src/pages/Details.jsx:51]: The delete button looks really nice here. Good choice.

**Review by scullenDev — APPROVED** [review]
Looks great; nice work, Liv!

**Pull request merged**